### PR TITLE
ENH: consistency of input args for boundaries -  pd.interval_range

### LINKED ISSUE #40245: ENH: consistency of input args for boundaries
There are some methods where boundary inputs are required.

This issue proposes standardising to the keyword arg: `inclusive` and using values `{"both", "neither", "left", "right"}`.

- [x] Styler.highlight_between():  `'inclusive' in {"both", "neither", "left", "right"}` 
- [x] Series.between() (#40628): `'inclusive' in {"both", "neither", "left", "right"}` 
- [x] pd.date_range() https://github.com/pandas-dev/pandas/pull/43504
- [x] DataFrame.between_time() https://github.com/pandas-dev/pandas/pull/43248
- [ ] pd.Interval https://github.com/pandas-dev/pandas/pull/46522
- [ ] pd.interval_range https://github.com/pandas-dev/pandas/pull/46355
- [ ] pd.IntervalIndex https://github.com/pandas-dev/pandas/pull/46522
- [ ] pd.cut
- [ ] Series.resample
- [ ] DataFrame.rolling(closed)

More methods yet to be identified.

### PR DESCRIPTION
**Pull request by weikhor**
- [x] xref #40245 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -973,7 +973,7 @@ def interval_range(
         for numeric and 'D' for datetime-like.
     name : str, default None
         Name of the resulting IntervalIndex.
-    closed : {'left', 'right', 'both', 'neither'}, default 'right'
+    inclusive : {'left', 'right', 'both', 'neither'}, default 'right'
```
Comment (jreback) [inline pandas/core/indexes/interval.py:976]: you need to actually deprecate this, not just change it

**Review by jreback — CHANGES_REQUESTED** [review]
deprecation means actually deprecating the paremeter & testing it

**Comment by jreback** [comment]
see the other PRs on that issue for how to do this.

File: pandas/core/indexes/interval.py (hunk 1 above)
Comment (weikhor) [reply pandas/core/indexes/interval.py]: Have update it.

File: pandas/core/indexes/interval.py [hunk 2]
```diff
@@ -977,6 +977,14 @@ def interval_range(
         Whether the intervals are closed on the left-side, right-side, both
         or neither.
 
+        .. deprecated:: 1.5.0
+           Argument `closed` has been deprecated to standardize boundary inputs.
```
Comment (jreback) [inline pandas/core/indexes/interval.py:981]: we need 1 tests to assert that `closed` is showing the proper deprecation message.
we need a test that asserts if *both* are passed there is an error raised

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/interval.py (hunk 2 above)
Comment (weikhor) [reply pandas/core/indexes/interval.py:983]: ok. Have updated.

Comment (weikhor) [reply pandas/core/indexes/interval.py:983]: Have updated. Done

**Comment by jreback** [comment]
> pandas/core/indexes/interval.py:1152: error: Argument "closed" to "from_breaks" of "IntervalIndex" has incompatible type "Optional[str]"; expected "str"  [arg-type]

File: pandas/core/indexes/interval.py [hunk 3]
```diff
@@ -1015,33 +1025,54 @@ def interval_range(
...
     intervals within the ``IntervalIndex`` are closed.
 
-    >>> pd.interval_range(end=5, periods=4, closed='both')
+    >>> pd.interval_range(end=5, periods=4, inclusive='both')
     IntervalIndex([[1, 2], [2, 3], [3, 4], [4, 5]],
                   dtype='interval[int64, both]')
     """
+    if inclusive is not None and not isinstance(closed, lib.NoDefault):
+        raise ValueError(
+            "Deprecated argument `closed` cannot be passed "
```
Comment (jreback) [inline pandas/core/indexes/interval.py:1057]: where is the test for this?

File: pandas/core/indexes/interval.py [hunk 4]
```diff
@@ -1015,33 +1025,54 @@ def interval_range(
...
+        warnings.warn(
+            "Argument `closed` is deprecated in favor of `inclusive`.",
+            FutureWarning,
+            stacklevel=2,
+        )
+        if closed is None:
+            inclusive = "both"
+        elif closed in ("both", "neither", "left", "right"):
+            inclusive = closed
+        else:
```
Comment (jreback) [inline pandas/core/indexes/interval.py:1070]: where are these tests for this?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/interval.py (hunk 4 above)
Comment (jreback) [reply pandas/core/indexes/interval.py:1072]: no you need a test to assert this

File: pandas/core/indexes/interval.py (hunk 3 above)
Comment (jreback) [reply pandas/core/indexes/interval.py:1059]: you need a test to assert this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/interval.py (hunk 4 above)
Comment (weikhor) [reply pandas/core/indexes/interval.py:1072]: ok. Have add test to assert this

File: pandas/core/indexes/interval.py (hunk 3 above)
Comment (weikhor) [reply pandas/core/indexes/interval.py:1059]: ok. Have add test to assert this

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @weikhor

**Pull request merged**